src/backend/distributed/worker/worker_merge_protocol.c

```diff
@@ -53,6 +57,49 @@ static void CopyTaskFilesFromDirectory(StringInfo schemaName, StringInfo relatio
 PG_FUNCTION_INFO_V1(worker_merge_files_into_table);
 PG_FUNCTION_INFO_V1(worker_merge_files_and_run_query);
 PG_FUNCTION_INFO_V1(worker_cleanup_job_schema_cache);
+PG_FUNCTION_INFO_V1(worker_create_schema);
+PG_FUNCTION_INFO_V1(worker_remove_jobdir);
+
+
+/*
+ * worker_create_schema creates a schema with the given job id in local.
+ */
+Datum
+worker_create_schema(PG_FUNCTION_ARGS)
+{
+	uint64 jobId = PG_GETARG_INT64(0);
+
+	StringInfo jobSchemaName = JobSchemaName(jobId);
+	bool schemaExists = false;
+	CheckCitusVersion(ERROR);
+
+	LockJobResource(jobId, AccessExclusiveLock);
+	schemaExists = JobSchemaExists(jobSchemaName);
+	if (!schemaExists)
+	{
+		CreateJobSchema(jobSchemaName);
+	}
+	UnlockJobResource(jobId, AccessExclusiveLock);
+
+	PG_RETURN_VOID();
+}
+
+
+/*
+ * worker_remove_jobdir removes the job directory with the given job id.
+ */
+Datum
+worker_remove_jobdir(PG_FUNCTION_ARGS)
+{
+	uint64 jobId = PG_GETARG_INT64(0);
+	StringInfo jobDirectoryName = JobDirectoryName(jobId);
+
+	CheckCitusVersion(ERROR);
+
+	CitusRemoveDirectory(jobDirectoryName);
```

Comment:
This is kind of dangerous, since any user can call it. We should check whether the user owns the schema.